pytorch_lightning/plugins/ddp_plugin.py

```diff
@@ -91,23 +92,26 @@ def init_ddp_connection(
                 torch_backend, rank=global_rank, world_size=world_size
             )
 
-    def on_before_forward(self, model: LightningModule, *args):
+    def on_before_forward(self, model: LightningDistributedDataParallel, *batch):
         """
         Override to handle custom input to device logic. For DDP, no logic is required as this is handled internally
         within the DDP wrapper.
 
         Example::
 
-            def on_before_forward(self, model, *args):
-                batch, batch_idx = args
+            def on_before_forward(self, model, *batch):
+                batch, batch_idx = batch
                 return batch.to(model.device)
 
         Args:
-            args: Inputs to the model.
+            batch: Inputs to the model.
             model: Model to train.
-        Returns: args moved to correct device if needed.
+        Returns: batch moved to correct device if needed.
         """
-        return args
+        if model.device_ids is not None and len(model.device_ids) == 1:
```

Comment:
This is a nice addition! Any reason why we can't make this a property `is_single_process_single_device`? I think it also clears things up and maybe even a comment:

```
# when using single process/single device move all objects to appropriate device as no scatter is necessary
```

Reply:
Done !